lib/rubocop/cop/style/conditional_assignment.rb

```diff
@@ -290,13 +288,11 @@ def autocorrect(node)
           end
         end
 
-        private
-
         def assignment_node(node)
           *_variable, assignment = *node
 
           if assignment.begin_type? && assignment.children.one?
-            assignment, = *assignment
```

Comment:
This was actually intentional for performance reasons (+ I find it more a bit more readable). We discourage the use of `children` across the codebase and rely heavily on destructuring nodes.